src/editor/sprite_editor/sprite_editor_clips.cpp

```
@@ -0,0 +1,317 @@
+void SpriteEditor::DrawClipsPane() {
+    ImGui::Text("Clips: %d", static_cast<int>(m_clips.size()));
+
+    // ---- Clip preview ----
+    if (m_selectedClip >= 0 && m_selectedClip < static_cast<int>(m_clips.size())) {
+        auto const& clip = m_clips[m_selectedClip];
+
+        // Advance animation playback
+        if (m_clipPlaying && !clip.desc.frames.empty()) {
+            m_clipElapsedMs += ImGui::GetIO().DeltaTime * 1000.0f;
+            while (m_clipPlaying) {
+                int const dur = std::max(clip.desc.frames[m_clipCurrentStep].durationMs, 1);
```

Clang (14.0.6)

[error] variable 'dur' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)